api/include/opentelemetry/common/macros.h

```diff
@@ -175,3 +175,24 @@ point.
 #  define OPENTELEMETRY_API_SINGLETON
 
 #endif
+
+//
+// The if/elif order matters here. If both OPENTELEMETRY_BUILD_IMPORT_DLL and
+// OPENTELEMETRY_BUILD_EXPORT_DLL are defined, the former takes precedence.
+//
+#if defined(_MSC_VER) && defined(OPENTELEMETRY_BUILD_IMPORT_DLL)
```

Comment:
I think we can declare `OPENTELEMETRY_EXPORT` just like `DLL_PUBLIC` in https://gcc.gnu.org/wiki/Visibility#How_to_use_the_new_C.2B-.2B-_visibility_support for better compatibility.
We meet some users add `-fvisibility=hidden` to `CXXFLAGS` environment variable before and it will prevent otel-cpp API to be shared between modules.
And also, add `__attribute__ ((visibility ("default")))` into declaration will also make RTTI codes public, or the virtual base classes can not be used by external codes when compiling otel-cpp with `-fvisibility=hidden`.

Reply:
Is this for supporting DLL with cygwin/gcc? I thought about this, but as there is some extra work needed to make it work like exporting the unreferenced symbols explicitly. As it is not been tested, probably I could leave a TODO item here?

Reply:
That's great.

Reply:
Added TODO to `macros.h`?